lxd/storage_buckets.go

```diff
@@ -55,6 +55,54 @@ var storagePoolBucketKeyCmd = APIEndpoint{
 	Put:    APIEndpointAction{Handler: storagePoolBucketKeyPut, AccessHandler: allowPermission(entity.TypeStorageBucket, auth.EntitlementCanEdit, "poolName", "bucketName")},
 }
 
+// storageBucketAccessHandler returns an access handler that checks for the given entitlement against a storage bucket.
+// The storage pool containing the bucket and the effective project of the bucket are added to the request context for
+// later use.
+func storageBucketAccessHandler(entitlement auth.Entitlement) func(d *Daemon, r *http.Request) response.Response {
+	return func(d *Daemon, r *http.Request) response.Response {
+		s := d.State()
+
+		projectName := request.ProjectParam(r)
+		bucketProjectName, err := project.StorageBucketProject(r.Context(), s.DB.Cluster, projectName)
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		request.SetCtxValue(r, request.CtxEffectiveProjectName, bucketProjectName)
+
+		poolName, err := url.PathUnescape(mux.Vars(r)["poolName"])
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		pool, err := storagePools.LoadByName(s, poolName)
+		if err != nil {
+			return response.SmartError(fmt.Errorf("Failed loading storage pool: %w", err))
+		}
+
+		request.SetCtxValue(r, ctxStoragePool, pool)
+
+		bucketName, err := url.PathUnescape(mux.Vars(r)["bucketName"])
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		// If the storage pool is a remote driver, the URL of the storage bucket will not contain a target parameter
```

Comment:
This comment seems misleading, a client can add a target parameter to any request it pleases, so to say a URL "will not" contain a target parameter in certain scenarios doesn't seem accurate.

Perhaps "should not" would be clearer?

Reply:
Ah yes, in the context of fine-grained auth, the unique URL for the resource does not have a target parameter because the volume does not have a location. The request URL might have a target parameter. I'll update the comment.

Reply:
Ah yes, in the context of fine-grained auth, the unique URL for the resource does not have a target parameter because the volume does not have a location. The request URL might have a target parameter. I'll update the comment.

Reply:
Ah yes, in the context of fine-grained auth, the unique URL for the resource does not have a target parameter because the volume does not have a location. The request URL might have a target parameter. I'll update the comment.